### ISSUE TITLE
Typechecking `types.py` does not work - only after you rename the file

### ISSUE THREAD

**Issue by hgiesel** [python/mypy#8351]
* Are you reporting a bug, or opening a feature request?
Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```
# types.py
import typing
import attr

@attr.s(auto_attribs=True, slots=True, frozen=True)
class Test:
    b: bool = False
```

* What is the actual behavior/output?
```
> mypy types.py
types.py:4: error: Cannot determine type of 's'
Found 1 error in 1 file (checked 1 source file)
```

* What is the behavior/output you expect?
```
> mypy types.py
Success: no issues found in 1 source file
```

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?
```
> mypy --version
mypy 0.761

> python3.8 --version
Python 3.8.1

> pip3.8 list
Package           Version
----------------- -------
attrs             19.3.0
mypy              0.761
mypy-extensions   0.4.3
pip               19.3.1
setuptools        42.0.2
typed-ast         1.4.1
typing-extensions 3.7.4.1
wheel             0.33.6
```

I tried it out with `mypy 0.770+dev.150d4921e702512e8afa0fc9cc22e71e720c3069`, and yes.

* What are the mypy flags you are using? (For example --strict-optional)
None.

* If mypy crashed with a traceback, please paste
  the full traceback below.
None.

It works as soon as your rename `types.py` to, e.g. `t.py`. I first asked this question on [Stack Overflow](https://stackoverflow.com/questions/59993074/mypy-does-not-recognize-attrs-cannot-determine-type-of-s/60014346#60014346) and a user brought this fact to my attention.

**Comment by JukkaL** [comment]
This is essentially a duplicate of #1876. It will be difficult to make mypy support this use case, so likely we'll just generate an error in this case.
